src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -145,19 +145,23 @@ protected override void PostPersist(ApplicationEngine engine)
                 OracleResponse response = tx.GetAttribute<OracleResponse>();
                 if (response is null) continue;
 
+                // Check tx state
+                if (engine.Snapshot.Transactions.TryGet(tx.Hash)?.VMState != VMState.HALT) continue;
```

Comment:
Need to remove this line.

Reply:
Why if the transaction was fault, should not be taken into account.